cirq-core/cirq/value/measurement_key_test.py

```diff
@@ -32,6 +41,13 @@ def test_nested_key():
     assert nested_key.name == 'key'
     assert nested_key.path == ('nested',)
 
+    qubits = (cirq.LineQubit(0), cirq.LineQubit(1))
+    nested_qubit_key = cirq.MeasurementKey(qubits=qubits, path=('nested',))
+
+    assert nested_qubit_key.name == ''
+    assert nested_qubit_key.path == ('nested',)
+    assert nested_qubit_key.qubits == qubits
+
 
 def test_eq_and_hash():
```

Comment:
In the current setup, the following pairs are equal:
```
MeasurementKey(name='m') == MeasurementKey(name='m', qubits=LineQubit.range(2))
MeasurementKey(name='0,1') == MeasurementKey(qubits=LineQubit.range(2))
```
The first one _might_ be okay, since the two should have similar behavior, but it feels fragile.
The second one is definitely not okay, since `with_qubits` will modify one but not the other.

Equality with strings can still be based on `str(key)` for convenience, but my sense is that equality between `MeasurementKey`s needs to be based on their entire contents to avoid unexpected behavior.

Reply:
Done.  Kept both cases distinct and added tests/documentation to clarify.